### PR TITLE
stackrox: fix nightly ocp 4.12 testing

### PR DESCRIPTION
**Pull request by BradLugo**
Splits nightly ocp 4.12 testing into its own variant and uses `cluster_profile` instead of `cluster_claim`. Prerequisite for https://github.com/openshift/release/pull/36771

/uncc
/hold

### CODE REVIEW

**Comment by BradLugo** [comment]
I guess the postsubmit jobs aren't rehearsable. Is there a way to test these changes?

/cc @gavin-stackrox 
/unhold

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-nightlies__ocp-4-12.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  steps:
+    cluster_profile: aws
+    env:
+      QA_TEST_TARGET: test
+      TEST_SUITE: openshift-4-operator-e2e-tests
+    test:
+    - ref: stackrox-e2e
+    workflow: ipi-aws
+  timeout: 5h0m0s
+- as: ui-e2e-tests
```
Comment (gavin-stackrox) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-nightlies__ocp-4-12.yaml:45]: UI e2e are not passing under OCP yet, so don't add these to nightlies (yet).

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]

**Comment by gavin-stackrox** [comment]
> I guess the postsubmit jobs aren't rehearsable. Is there a way to test these changes?

You _could_ remove the `postsubmit: true` to test but because these are a duplicate of what we know works for master, I don't think that is necessary.

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-nightlies__ocp-4-12.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    test:
+    - ref: stackrox-e2e
+    workflow: ipi-aws
+  timeout: 5h0m0s
+- as: operator-e2e-tests
+  postsubmit: true
+  steps:
+    cluster_profile: aws
+    env:
+      QA_TEST_TARGET: test
```
Comment (gavin-stackrox) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-nightlies__ocp-4-12.yaml:39]: Only used by qa-e2e.
```suggestion
```

**Review by gavin-stackrox — APPROVED** [review]
/lgtm

**Pull request merged**